app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModel.kt

```diff
@@ -3,23 +3,38 @@ package com.intive.picover.parties.viewmodel
 import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.intive.picover.parties.repository.MockedPartiesRepository
 import com.intive.picover.parties.state.PartiesState
+import com.intive.picover.parties.state.PartiesState.Error
+import com.intive.picover.parties.state.PartiesState.Loaded
+import com.intive.picover.parties.state.PartiesState.Loading
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
+import kotlinx.coroutines.launch
 
 @HiltViewModel
 class PartiesViewModel @Inject constructor(
 	private val partiesRepository: MockedPartiesRepository,
 ) : ViewModel() {
 
-	val parties: MutableState<PartiesState> = mutableStateOf(PartiesState.Loading)
+	val parties: MutableState<PartiesState> = mutableStateOf(Loading)
 
 	init {
-		loadParties()
+		viewModelScope.launch {
+			loadParties()
+		}
 	}
 
-	private fun loadParties() {
-		parties.value = PartiesState.Loaded(partiesRepository.getParties())
+	private suspend fun loadParties() {
```

Comment:
good question how we would like to handle this in viewmodels, we can do it like this, OR:   

Use flow here, and for example in tests, test happy path, with first and second flow value
and test error path with first and second value of flow